sale_commission/wizard/wizard_settle.py

```diff
@@ -85,11 +85,15 @@ def action_settle(self):
                         settlement = settlement_obj.create(
                             {'agent': agent.id,
                              'date_from': sett_from,
-                             'date_to': sett_to})
+                             'date_to': sett_to,
+                             'company_id':
+                                agent_lines[pos].invoice_line.company_id.id})
```

Comment:
There are multiple valid uses in multi-company environment outside admin. For example, parent companies with access to several child companies, so your code must be aware to this. 

There's already enough code in Odoo core that doesn't take this into account to add more.